src/toolbars/TimeSignatureToolBar.cpp

```diff
@@ -228,5 +249,35 @@ AttachedToolBarMenuItem sAttachment{
       for selecting a time range of audio */
    TimeSignatureToolBar::ID(), wxT("ShowTimeSignatureTB"), XXO("Time Signature Toolbar")
 };
+
+// Undo/redo handling of time signature changes
+// DV: where should this really go?
```

Comment:
Answering that question, I have suggested that a lib-time-signature, or lib-beats or whatever you name it, would make sense as a place in which to put registrations that push more behavior into the other libraries.

It was true before this commit that all `UndoStateExtension` subclasses and their factories were in green libraries, not in Audacity.  Given lib-beats, that could remain so.